aries_cloudagent/protocols/endorse_transaction/v1_0/manager.py

```diff
@@ -0,0 +1,440 @@
+from .messages.endorsed_transaction_response import EndorsedTransactionResponse
+from .messages.refused_transaction_response import RefusedTransactionResponse
+from .messages.cancel_transaction import CancelTransaction
+from .messages.transaction_resend import TransactionResend
+from .messages.messages_attach import MessagesAttach
+
+from ....ledger.base import BaseLedger
+from ....core.error import BaseError
+from ....core.profile import Profile, ProfileSession
+from ....ledger.error import LedgerError
+
+from ....wallet.base import BaseWallet
+
+
+class TransactionManagerError(BaseError):
+    """Transaction error."""
+
+
+class TransactionManager:
+    """Class for managing transactions."""
+
+    def __init__(self, session: ProfileSession, profile: Profile):
+        """
+        Initialize a TransactionManager.
+
+        Args:
+            session: The Profile Session for this transaction manager
+            profile: The Profile for this transaction manager
+        """
+        self._session = session
+        self._profile = profile
+        self._logger = logging.getLogger(__name__)
+
+    @property
+    def session(self) -> ProfileSession:
+        """
+        Accessor for the current Profile Session.
+
+        Returns:
+            The Profile Session for this transaction manager
+
+        """
+        return self._session
+
+    @property
+    def profile(self) -> Profile:
+        """
+        Accessor for the current Profile.
+
+        Returns:
+            The Profile for this transaction manager
+
+        """
+        return self._profile
+
+    async def create_request(
+        self,
+        connection_id: str = None,
+        expires_time: str = None,
+        transaction_message: dict = {},
+    ):
+        """
+        Create a new Transaction Record and Request.
+
+        Args:
+            connection_id: The connection id related to this transaction record
+            expires_time: The time till which the endorser should endorse the transaction
+            transaction_message: The actual data in the transaction payload
+
+        Returns:
+            The transaction Record and transaction request
+
+        """
+
+        wallet: BaseWallet = self.session.inject(BaseWallet, required=False)
+        if not wallet:
+            raise web.HTTPForbidden(reason="No wallet available")
+        author_did_info = await wallet.get_public_did()
+        if not author_did_info:
+            raise web.HTTPForbidden(reason="Public DID not found in wallet")
+        author_did = author_did_info.did
+        author_verkey = author_did_info.verkey
+
+        ledger: BaseLedger = self.session.inject(BaseLedger, required=False)
+
+        if not ledger:
+            reason = "No indy ledger available"
+            if not self.session.settings.get_value("wallet.type"):
+                reason += ": missing wallet-type?"
+            raise web.HTTPForbidden(reason=reason)
+
+        async with ledger:
```

Comment:
Copying my feedback over from the previous PR:

@HarshMultani @swcurran @ankita-p17 ... I’ve been doing some thinking about the design of the Endorser api. I think that the current approach (creating a signed transaction and dealing with TAA inside the api) is not the best. The existing api’s that write to the ledger (create schema, create cred def, etc.) already have all this logic, so adding it to the Endorser API is duplication.

I think to keep separation of concerns and to keep the code DRY:
• Update the ledger-updating api's (create schema, create cred def etc.) to either write directly to the ledger, or return a prepared (signed & TAA-accepted) transaction
• The controller can then pass this transaction as an attachment to the /transactions/request-endorsement endpoint
• (this endpoint then doesn't have to worry about anything other than packaging the attachment and sending to the Endorser for endorsement)
• Once the endorsement is received, the controller can call another endpoint (e.g. /transactions/write-to-ledger) to write the endorsed transaction to the ledger
• (This last step will involve some indy-specific work, adding the endorser signature to the indy transaction)

Since we already have code for preparing the indy transactions, I don't think we should add this to the endorser api's at all